docs/open-source/agent-factory.mdx

```diff
@@ -5,29 +5,31 @@ description: 'How to link a custom agent to your app'
 
 # Agent Factories
 
-Agent factories specify which agents are available to your app. In order to connect an agent to your app, you must first define an agent factory. To do so, subclass the [`AgentFactory`](https://github.com/vocodedev/vocode-python/blob/main/vocode/streaming/agent/factory.py) class to specify how agents are created. Here you can import and use your own custom agents.
+Agent factories specify which agents are available to your app. In order to connect an agent to your app, you must first define an agent factory. To do so, subclass the [`AbstractAgentFactory`](https://github.com/vocodedev/vocode-python/blob/main/vocode/streaming/agent/abstract_factory.py) class to specify how agents are created. Here you can import and use your own custom agents.
 
 ## Example
 
-First define your `AgentFactory`:
+First define your `AgentFactory`. In this example, we are creating a factory for a new type of agent called MyActionAgent:
 
 ```python
-from vocode.streaming.agent.factory import AgentFactory
+from vocode.streaming.agent.abstract_factory import AbstractAgentFactory
 
-class MyAgentFactory(AgentFactory):
-    def __init__(self, agent_config: AgentConfig, action_factory: MyActionFactory):
-        self.agent_config = agent_config
+class MyAgentFactory(AbstractAgentFactory):
+    def __init__(self, action_factory: MyActionFactory):
         self.action_factory = action_factory
 
     def create_agent(
         self, agent_config: AgentConfig, logger: Optional[logging.Logger] = None
     ) -> BaseAgent:
         if agent_config.type == "MY_ACTION":
             return MyActionAgent(
-                agent_config=typing.cast(ActionAgentConfig, self.agent_config),
+                agent_config=agent_config,
                 action_factory=self.action_factory
             )
-        raise Exception("Invalid agent config")
+        elif agent_config.type == "other_agent_type":
+            ...
+        else:
+            raise Exception("Invalid agent config")
```

Comment:
bottom AgentFactory instantiation needs to be updated

Reply:
No this was intentional (as in they can add others). Would it be better to have just the one agent type?

Reply:
like it no longer can take in an `agent_config`, instantiation needs to be updated